src/content/app/tools/blast/views/blast-submission-results/BlastSubmissionResults.tsx

```diff
@@ -33,7 +46,119 @@ const BlastSubmissionResults = () => {
 };
 
 const Main = () => {
-  return <div>Here will be submission details</div>;
+  const { submissionId } = useParams() as { submissionId: string };
+  const blastSubmission = useAppSelector((state) =>
+    getBlastSubmissionById(state, submissionId)
+  );
+
+  if (!blastSubmission) {
+    return null;
+  }
+
+  const { submittedData, results } = blastSubmission;
+
+  const resultsGroupedBySequence = submittedData.sequences.map((sequence) => {
+    const blastResults = results.filter((r) => r.sequenceId === sequence.id);
+    return {
+      sequence,
+      species: submittedData.species,
+      blastResults
+    };
+  });
+
+  const sequenceBoxes = resultsGroupedBySequence.map((data) => (
+    <SequenceBox
+      key={data.sequence.id}
+      species={data.species}
+      sequence={data.sequence}
+      blastResults={data.blastResults}
+    />
+  ));
+
+  return (
+    <div className={styles.blastSubmissionResultsContainer}>
+      <BlastSubmissionHeader
+        submission={blastSubmission}
+        isAnyJobRunning={false}
+      />
+      {sequenceBoxes}
+    </div>
+  );
+};
+
+type SequenceBoxProps = {
+  sequence: {
+    id: number;
+    value: string;
+  };
+  species: Species[];
+  blastResults: BlastResult[];
+};
+
+const SequenceBox = (props: SequenceBoxProps) => {
+  const { sequence, species, blastResults } = props;
+
+  return (
+    <>
+      <div key={sequence.id} className={styles.sequenceBoxWrapper}>
+        <div className={styles.resultsSummaryRow}>
+          <div className={styles.sequenceId}>Sequence {sequence.id}</div>
+          <div className={styles.sequenceHeader}>
+            {'>' + (parseBlastInput(sequence.value)[0].header || '')}
```

Comment:
Don't think `undefined` is rendered by React so there's no need for an empty string to replace it

Reply:
This is what would happen without defaulting to a string:

![image](https://user-images.githubusercontent.com/6834224/182549001-11fcdc5a-7159-4d0c-bdb4-684243241d60.png)

@jyothishnt I am still seeing this on the review deployment

Reply:
Ah is this because this is a string concatenation?